raiden/transfer/channel.py

```diff
@@ -1574,6 +1639,50 @@ def send_lock_expired(
     return []
 
 
+def is_withdraw_expired(
+    block_number: BlockNumber, expiration_threshold: BlockNumber
+) -> SuccessOrError:
+    """ Determine whether a withdraw has expired.
+
+    The withdraw has expired if the current block exceeds
+    the withdraw's expiration + confirmation blocks.
+    """
+    if block_number < expiration_threshold:
+        msg = (
+            f"current block number ({block_number}) is not larger than "
+            f"withdraw.expiration + confirmation blocks ({expiration_threshold})"
+        )
+        return (False, msg)
+
+    return (True, None)
+
+
+def events_for_expired_withdraws(
+    channel_state: NettingChannelState,
+    block_number: BlockNumber,
+    pseudo_random_generator: random.Random,
+) -> List[SendWithdrawExpired]:
+    events: List[SendWithdrawExpired] = list()
+
+    for withdraw_state in channel_state.our_state.withdraws:
```

Comment:
Aren't the withdraws ordered? A later withdraw will have a higher expiration block, if the first is not expired then nonce of the ones following will be expired. Again, a datastructure for this would not hurt.

Reply:
Added a break on the first non-expired withdraw